Serhii Lakhman/Task 2/CAnyType.inl

```diff
@@ -0,0 +1,206 @@
+#include "CAnyType.h"
+
+
+template<class T>
+inline CAnyType<T>::CAnyType()
+{
+	this->Variable = nullptr;
+}
+
+
+
+template<class T>
+inline CAnyType<T>::CAnyType(T varible)
+{
+
+	this->Variable = new T;
+
+	*(this->Variable) = varible;
+	
+	TypeName = typeid(varible).name();
+
+}
+
+
+
+template<class T>
+inline CAnyType<T>::CAnyType(const CAnyType<T>& other)
+{
+	destroy();
+
+	 this->Variable = new T;
+
+	*(this->Variable) = *(other.Variable);
+
+}
+
+
+
+template<class T>
+inline CAnyType<T>::~CAnyType()
+{
+	destroy();
+}
+
+
+
+
+template<class T>
+inline void CAnyType<T>::destroy()
+{
+	delete this->Variable;
+
+	this->Variable = nullptr;
+}
+
+
+
+template<class T>
+inline void CAnyType<T>::MySwap(CAnyType<T>& FirstObject, CAnyType<T>& SecondObject)
+{
+	CAnyType<T> temp = move(FirstObject);
+	FirstObject = move(SecondObject);
+	SecondObject = move(temp);
+
+}
+
+
+
+
+template<class T>
+inline string CAnyType<T>::GetType()
+{
+	return this->TypeName;
+}
+
+
+
+
+template<class T>
+inline CAnyType<T>& CAnyType<T>::operator=(const CAnyType<T>& other)
+{
+	
+	if (this == &other)			return *this;
+	
+	destroy();
+
+	this->Variable = new T;
+
+	*(this->Variable) = *(other.Variable);
+
+	return *this;
+
+}
+
+
+
+template<class T1>
+inline ostream& operator<<(ostream& output, const CAnyType<T1>& Varible)
+{
+
+	output << *(Varible.Variable);
+
+	return output;
+
+}
+
+
+
+template<class T>
+inline CAnyType<T>& CAnyType<T>::operator=(T const Variable)
+{
+
+		destroy();
+
+		this->Variable = new T;
+
+		*(this->Variable) = Variable;
+
+		return *this;
+	
+}
+
+
+
+
+template<class T>
```

Comment:
it is redundant to write template twice

Reply:
If I want to accept two different data types, I need to announce another template or I think wrong?